core/src/main/java/org/jdbi/v3/core/mapper/MapEntryMapper.java

```diff
@@ -0,0 +1,164 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.mapper;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.lang.reflect.TypeVariable;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Map;
+import java.util.Optional;
+import org.jdbi.v3.core.config.ConfigRegistry;
+import org.jdbi.v3.core.config.JdbiConfig;
+import org.jdbi.v3.core.statement.StatementContext;
+
+import static com.google.common.collect.Maps.immutableEntry;
+import static java.util.Objects.requireNonNull;
+import static org.jdbi.v3.core.generic.GenericTypes.getErasedType;
+import static org.jdbi.v3.core.generic.GenericTypes.resolveType;
+
+/**
+ * Maps rows to {@link Map.Entry Map.Entry&lt;K, V&gt;}, provided there are mappers registered for types K and V. This mapper is registered out of the box.
+ *
+ * @param <K> entry key type
+ * @param <V> entry value type
+ */
+public class MapEntryMapper<K, V> implements RowMapper<Map.Entry<K, V>> {
+    private static final TypeVariable<Class<Map.Entry>> KEY_PARAM;
+    private static final TypeVariable<Class<Map.Entry>> VALUE_PARAM;
+
+    static {
+        TypeVariable<Class<Map.Entry>>[] mapParams = Map.Entry.class.getTypeParameters();
+        KEY_PARAM = mapParams[0];
+        VALUE_PARAM = mapParams[1];
+    }
+
+    private final RowMapper<K> keyMapper;
+    private final RowMapper<V> valueMapper;
+
+    @SuppressWarnings("unchecked")
+    static RowMapperFactory factory() {
+        return (type, config) -> {
+            if (type instanceof ParameterizedType && getErasedType(type).equals(Map.Entry.class)) {
+                Type keyType = resolveType(KEY_PARAM, type);
+                Type valueType = resolveType(VALUE_PARAM, type);
+
+                RowMapper<?> keyMapper = getKeyMapper(keyType, config);
+                RowMapper<?> valueMapper = getValueMapper(valueType, config);
+
+                return Optional.of(new MapEntryMapper(keyMapper, valueMapper));
+            }
+            return Optional.empty();
+        };
+    }
+
+    private static RowMapper<?> getKeyMapper(Type keyType, ConfigRegistry config) {
+        String column = config.get(Config.class).getKeyColumn();
+        if (column == null) {
+            return config.get(RowMappers.class)
+                    .findFor(keyType)
+                    .orElseThrow(() -> new NoSuchMapperException("No row mapper registered for map key " + keyType));
+        } else {
+            return config.get(ColumnMappers.class)
+                    .findFor(keyType)
+                    .map(mapper -> new SingleColumnMapper<>(mapper, column))
+                    .orElseThrow(() -> new NoSuchMapperException("No column mapper registered for map key " + keyType + " in column " + column));
+        }
+    }
+
+    private static RowMapper<?> getValueMapper(Type keyType, ConfigRegistry config) {
```

Comment:
I think it would be better to name the variable `valueType`, because we are looking for a mapper for the value column.